utensor_cgen/backend/utensor/code_generator/rearch/_code_generator.py

```diff
@@ -80,7 +82,202 @@ def _find_required_ops(self, ugraph):
           OperatorFactory.get_opertor(op_info)
         )
     return ops, placeholders, tensor_var_map
+
+  def _try_get_opt_snippets(self, ugraph, ops, tensor_var_map):
+    # snippets to be rendered in the generated function body
+    local_snippets = []
+    # snippets to be rendered in the weight header file
+    weight_snippets = []
+    # op -> op variable name
+    ops_map = {}
+    for i, op in enumerate(ops):
+      op_var_name = 'op_{:03d}'.format(i)
+      ops_map[op] = op_var_name
+      # all ops will be declared first
+      local_snippets.append(
+        op.get_declare_snippet(op_var_name, tensor_var_map)
+      )
+    out_tensor_var_names = [
+      tensor_var_map[tensor.name] for tensor in chain(*[
+        ugraph.ops_info[op_name].output_tensors
+        for op_name in ugraph.output_nodes
+      ])
+    ]
+
+    # dict: tensor name -> TimeslotAllocation
+    tensor_timeslots = ugraph.attributes.get(TopoOrderTensorTimeslotPlanner.KWARGS_NAMESCOPE)
```

Comment:
Should there be a try catch around this? What if the ugraph doesnt contain a TimeslotPlanner?

Reply:
That's why I use `get`
The `get` method will return a default value, which is `None` in this case, if the key is not in the dictionary.
If you look at the next line, I do have a check on the returned value and if there is no such plan available, it will roll back to naive code generation.